Assets/MixedRealityToolkit.Extensions/Sharing/SpatialAlignment/Common/ISpatialCoordinate.cs

```diff
@@ -85,4 +81,9 @@ public interface ISpatialCoordinate
         /// </summary>
         Quaternion CoordinateToWorldSpace(Quaternion quaternion);
     }
+
+    public interface ISpatialCoordinate<TIdentifier> : ISpatialCoordinate
+    {
+        TIdentifier Id { get; }
```

Comment:
Is there a super strong preference not to have everything string? I realize this means conversion back and forth, but shouldn't be too bad?

Reply:
The way I thought about this:
1. From within the implementation of a SpatialCoordianteSystem, the identifier has some concrete meaning and concrete type. That type is probably not a string in most cases, and can be strongly-typed.
2. From outside the implementation of a SpatialCoordinateSystem, the identifier is pretty meaningless and probably shouldn't be used for a purpose, unless I'm missing some purpose. It may or may not uniquely identify that coordinate within the same SpatialCoordinateSystem, and it certainly doesn't globally-uniquely identify a coordinate.

Given those two things are true, it made more sense to me to make the identifier a concrete value, and make it an optional thing that doesn't necessarily need to be implemented to have an ISpatialCoordinate.